src/util/tseries.h

```diff
@@ -0,0 +1,146 @@
+#pragma once
+
+#include <algorithm>
+#include <cstdint>
+#include <cstring>
+#include <vector>
+
+#include "stats.h"
+#include "types.h"
+
+namespace bpftrace::util {
+
+std::pair<std::vector<uint8_t>, uint64_t> reduce_tseries_value(
+    const std::vector<uint8_t> &values,
+    int nvalues,
+    const SizedType &value_type,
+    TSeriesAggFunc agg);
+
+template <typename T>
+std::pair<T, uint64_t> reduce_tseries_value(const std::vector<uint8_t> &values,
```

Comment:
just curious, why are `reduce_tseries_value` and `tseries_bounds` implemented in the header?